robot_interaction/src/robot_interaction.cpp

```diff
@@ -116,9 +116,28 @@ void RobotInteraction::addActiveComponent(const InteractiveMarkerConstructorFn &
   active_generic_.push_back(g);
 }
 
+static const double DEFAULT_SCALE = 0.25;
+double RobotInteraction::computeLinkMarkerSize(const std::string &group, const std::string &link)
+{
+  const robot_model::JointModelGroup *jmg = robot_model_->getJointModelGroup(group);
+  if (!jmg) return DEFAULT_SCALE;
+
+  robot_state::RobotState default_state(robot_model_);
+  default_state.setToDefaultValues();
+
+  const robot_model::LinkModel *lm = default_state.getLinkModel(link);
+  if (!lm) return DEFAULT_SCALE;
+
+  const Eigen::Vector3d &ext = lm->getShapeExtentsAtOrigin();
+  // slightly bigger than the link diameter
+  double s = 1.01 * ext.norm();
+
+  // clip scale to 5cm
+  return std::max(0.05, s);
```

Comment:
how did you choose 5cm here? i've seen some robots whose model does not properly follow metric sizing, i.e. they could be a couple of orders of magnitude larger or smaller than real life meters. Seems this could be error prone hard coding 0.05